src/main/cpp/checkpoint/CheckPoint.cpp

```diff
@@ -0,0 +1,600 @@
+#include <pop/Person.h>
+#include <util/Errors.h>
+#include <util/InstallDirs.h>
+
+#include <cstring>
+#include <fstream>
+#include <sstream>
+#include <streambuf>
+#include <string>
+#include <vector>
+#include <boost/filesystem.hpp>
+
+using namespace std;
+
+namespace stride {
+namespace checkpoint {
+
+CheckPoint::CheckPoint(const std::string& filename, unsigned int interval) : m_filename(filename), m_limit(interval) {}
+
+void CheckPoint::CreateFile()
+{
+	m_file = H5Fcreate(m_filename.c_str(), H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT);
+	H5Fclose(m_file);
+}
+
+void CheckPoint::CloseFile() { H5Fclose(m_file); }
+
+void CheckPoint::WriteConfig(const SingleSimulationConfig& conf)
+{
+	htri_t exist = H5Lexists(m_file, "Config", H5P_DEFAULT);
+	if (exist <= 0) {
+		hid_t temp = H5Gcreate2(m_file, "Config", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+		H5Gclose(temp);
+	}
+
+	std::shared_ptr<CommonSimulationConfig> common_config = conf.common_config;
+	std::shared_ptr<LogConfig> log_config = conf.log_config;
+
+	WriteFileDSet(common_config->disease_config_file_name, "disease");
+	WriteFileDSet(common_config->contact_matrix_file_name, "contact");
+	WriteFileDSet(conf.GetPopulationPath(), "popconfig");
+	if (conf.GetGeodistributionProfilePath().size() != 0) {
+		WriteFileDSet(conf.GetGeodistributionProfilePath(), "geoconfig");
+	}
+	if (conf.GetReferenceHouseholdsPath().size() != 0) {
+		WriteFileDSet(conf.GetReferenceHouseholdsPath(), "household");
+	}
+
+	hid_t group = H5Gopen2(m_file, "Config", H5P_DEFAULT);
+	hsize_t dims = 2;
+	hid_t dataspace = H5Screate_simple(1, &dims, NULL);
+	hbool_t bools[2];
+	bools[0] = common_config->track_index_case;
+	bools[1] = log_config->generate_person_file;
+	hid_t attr = H5Acreate2(group, "bools", H5T_IEEE_F64LE, dataspace, H5P_DEFAULT, H5P_DEFAULT);
+	H5Awrite(attr, H5T_NATIVE_HBOOL, bools);
+	H5Sclose(dataspace);
+	H5Aclose(attr);
+
+	dims = 4;
+	dataspace = H5Screate_simple(1, &dims, NULL);
+	attr = H5Acreate2(group, "uints", H5T_IEEE_F64LE, dataspace, H5P_DEFAULT, H5P_DEFAULT);
+	unsigned int uints[4];
+	uints[0] = common_config->rng_seed;
+	uints[1] = common_config->number_of_days;
+	uints[2] = common_config->number_of_survey_participants;
+	uints[3] = (unsigned int)log_config->log_level;
+	H5Awrite(attr, H5T_NATIVE_UINT, uints);
+	H5Sclose(dataspace);
+	H5Aclose(attr);
+
+	dims = 3;
+	dataspace = H5Screate_simple(1, &dims, NULL);
+	attr = H5Acreate2(group, "doubles", H5T_IEEE_F64LE, dataspace, H5P_DEFAULT, H5P_DEFAULT);
+	double doubles[3];
+	doubles[0] = common_config->r0;
+	doubles[1] = common_config->seeding_rate;
+	doubles[2] = common_config->immunity_rate;
+
+	H5Awrite(attr, H5T_NATIVE_DOUBLE, doubles);
+	H5Sclose(dataspace);
+	H5Aclose(attr);
+
+	dims = log_config->output_prefix.size();
+	dataspace = H5Screate_simple(1, &dims, NULL);
+	attr = H5Acreate2(group, "prefix", H5T_IEEE_F64LE, dataspace, H5P_DEFAULT, H5P_DEFAULT);
+	char prefix[log_config->output_prefix.size()];
```

Comment:
Can't you just `H5Awrite(attr, HST_NATIVE_CHAR, log_config->output_prefix.c_str()`?